src/components/Proposal/Votes/index.tsx

```diff
@@ -22,23 +28,144 @@ import {
 } from '../styles';
 
 import { useContext } from 'contexts';
-import { bnum, toPercentage } from 'utils';
+import { bnum, toPercentage, verifySignedVote } from 'utils';
+import { utils } from 'ethers';
+import useJsonRpcProvider from 'hooks/Guilds/web3/useJsonRpcProvider';
 
 const Votes = () => {
   const {
-    context: { daoStore, providerStore, daoService },
+    context: {
+      configStore,
+      daoStore,
+      providerStore,
+      daoService,
+      messageLoggerService,
+      orbitDBService,
+    },
   } = useContext();
 
   //State
-  const [decision, setDecision] = useState(null);
+  const [signVote, setSignVote] = useState(false);
+  const [decision, setDecision] = useState(0);
   const [votePercentage, setVotePercentage] = useState(0);
+  const [signedVotesOfProposal, setSignedVotesOfProposal] = useState([]);
+  const [loadingSignedRinkebyVotes, setLoadingSignedRinkebyVotes] =
+    useState(true);
+  const [loadingSignedOrbitDBVotes, setLoadingSignedOrbitDBVotes] =
+    useState(true);
 
   // We should get the ID in another way
   const proposalId = useLocation().pathname.split('/')[3];
 
   const proposal = daoStore.getProposal(proposalId);
   const proposalEvents = daoStore.getProposalEvents(proposalId);
   const { account } = providerStore.getActiveWeb3React();
+  const signedVoteMessageId = utils.id(`dxvote:${proposalId}`);
+  const { finishTime } = daoStore.getProposalStatus(proposalId);
+  const votingParameters = daoStore.getVotingParametersOfProposal(proposalId);
+  const votingMachineAddress = daoStore.getVotingMachineOfProposal(proposalId);
+  const finishTimeReached = finishTime.toNumber() < moment().unix();
+  const isDXDVotingMachine =
+    configStore.getNetworkContracts().votingMachines.dxd &&
+    configStore.getNetworkContracts().votingMachines.dxd.address ==
+      votingMachineAddress;
+
+  const rinkebyProvider = useJsonRpcProvider(4);
+
+  messageLoggerService
+    .getMessages(signedVoteMessageId, rinkebyProvider)
+    .then(messagesEvents => {
+      console.debug('[Rinkeby Proposal messages]', messagesEvents);
+      messagesEvents.map(messagesEvent => {
+        if (
+          messagesEvent.args &&
+          messagesEvent.args.message.length > 0 &&
+          messagesEvent.args.message.split(':').length > 6
+        ) {
+          const signedVote = messagesEvent.args.message.split(':');
+          const validSignature = verifySignedVote(
+            signedVote[1],
+            signedVote[2],
+            signedVote[3],
+            signedVote[4],
+            signedVote[5],
+            signedVote[6]

```

Comment:
it would be helpful if you added a quick comment on what these represent exactly